signature-specification.md

```diff
@@ -0,0 +1,230 @@
+# Signature Specification
+
+This document describes how Notary v2 signatures are created and stored.
+The document consists of following sections:
+* **[Storage](#storage)**: Describes how signatures are stored in OCI registry
+* **[Signature Envelope](#signature-envelope)**: Describes how signatures are created.
+
+
+## Storage
+This section describes how Notary v2 signatures are stored in the OCI Distribution conformant registry.
+Notary v2 uses [ORAS artifact manifest](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md) to store the signature in the repository. The signature artifact manifest consists of a signature type, a reference to the manifest of the artifact being signed, a blob referencing to the signature, and a collection of annotations.
+
+-<img src="media/signature-specification.jpg" width="550" height="450">
+
+- **`artifactType`**: Used to identify the Notary signature artifact. The supported value is `application/vnd.cncf.notary.v2`.
+- **`blobs`**: A collection consisting of only one [artifact descriptor](https://github.com/oras-project/artifacts-spec/blob/main/descriptor.md) that refers to a signature envelope.
+   - **`mediaType`**: Defines the media type of signature envelope blob. The supported value is `application/jose+json`
```

Comment:
For the media type `application/jose+json`, will `application/vnd.cncf.notary.signature.v2.jws+json` be better?

Reply:
Added `application/jose+json` because its already registered in [iana](https://www.iana.org/assignments/media-types/media-types.xhtml). Also, its actually an serialized JWS JSON signature.

Reply:
After re-reading this, the `mediaType` hers is the blob of the signature. We had similar feedback in OCI Artifact conversations to use existing `mediaTypes` when possible. So, `application/jose~json` seems good, particularly as it's already registered with iana.org